File: js/179_LargestNumber.js
/**
 * @param {number[]} nums
 * @return {string}
 */
var largestNumber = function(nums) {
    var ret = "";
    var tmp = [];
    for(var i = 0; i < nums.length; ++i){
        tmp.push(nums[i].toString())
    }
    tmp.sort(function(a, b){
        return (a + b) > (b + a) ?  -1 : 1;
    });
    if(tmp[0] === "0") return "0";
    for(var j = 0; j < tmp.length; ++j){
        ret += tmp[j];
    }
    return ret;
};

File: cpp/179_LargestNumber.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        string ret = "";
        vector<string> tmp;
        for(auto i : nums){
            tmp.push_back(std::to_string(i));
        }
        std::sort(tmp.begin(), tmp.end(), compareString);
        if(tmp[0] == "0") return "0";
        for(auto i : tmp){
            ret += i;
        }
        return ret;
    }
    
    static bool compareString(string a, string b){
        return a+b > b+a;
    }
};